File: 8 longest consecutive sequence.js
/*
Given an unsorted array of integers nums, return the length of the longest consecutive elements sequence.
You must write an algorithm that runs in O(n) time.
*/

var longestConsecutive = function (nums) {
    if (nums == null || nums.length === 0) return 0;
  
    const set = new Set(nums);
    let max = 0;
  
    for (let num of set) {
//ignore later values in a sequence
      if (set.has(num - 1)) continue; 
  
      let currNum = num;
      let currMax = 1;
  
      while (set.has(currNum + 1)) {
        currNum++;
        currMax++;
      }
      max = Math.max(max, currMax);
    }
  
    return max;
  }
  };

File: 1 arrays and hash tables/8 longest consecutive sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        Lset = set(nums)
        Smax = 0
        for num in Lset:
            if num-1 in Lset:
                continue
            currNum=num
            currMax=1
            while currNum+1 in Lset:
                currMax += 1
                currNum += 1
            Smax = max(Smax, currMax)
        return Smax